In [1]:
import pandas as pd

# Dataset 1 - Weather

### Q1
Load your dataset into python using the pandas library and save data into a dataframe named dfi (where i is one of your datasets, for a total of 4).

In [2]:
df1 = pd.read_csv('https://github.com/data602sps/datasetspractice/raw/main/weather.csv')

### Q2
Preview your data by calling your dataframe’s name. How many columns and rows do you see?

In [3]:
df1

Unnamed: 0 origin  year  month   day  hour   temp   dewp  humid  \
0              1    EWR  2013    1.0   1.0   0.0  37.04  21.92  53.97   
1              2    EWR  2013    1.0   1.0   1.0  37.04  21.92  53.97   
2              3    EWR  2013    1.0   1.0   2.0  37.94  21.92  52.09   
3              4    EWR  2013    1.0   1.0   3.0  37.94  23.00  54.51   
4              5    EWR  2013    1.0   1.0   4.0  37.94  24.08  57.04   
...          ...    ...   ...    ...   ...   ...    ...    ...    ...   
8714        8715    JFK  2013    9.0   2.0  20.0  75.20  73.40  94.14   
8715        8716    JFK  2013   10.0  23.0  10.0  48.92  39.02  68.51   
8716        8717    JFK  2013   10.0  23.0  11.0  48.92  39.02  68.51   
8717        8718    JFK  2013   12.0  17.0   5.0  26.96  10.94  50.34   
8718        8719    LGA  2013    8.0  22.0  22.0  75.92  66.92  73.68   

      wind_dir  wind_speed  wind_gust  precip  pressure  visib  
0        230.0    10.35702  11.918651     0.0    1013.9   10.0  
1        230.0    13.80936  15.891535     0.0    1013.0   10.0  
2        230.0    12.65858  14.567241     0.0    1012.6   10.0  
3        230.0    13.80936  15.891535     0.0    1012.7   10.0  
4        240.0    14.96014  17.215830     0.0    1012.8   10.0  
...        ...         ...        ...     ...       ...    ...  
8714     200.0     4.60312   5.297178     0.0       NaN    4.0  
8715      60.0     4.60312   5.297178     0.0    1008.1   10.0  
8716      40.0     4.60312   5.297178     0.0    1008.5   10.0  
8717      40.0     4.60312   5.297178     0.0    1023.9   10.0  
8718     210.0     8.05546   9.270062     0.0    1011.9   10.0  

[8719 rows x 15 columns]

JupyterLab truncates the dataframe, showing the first 5 and last 5 rows. However, it does display all 15 columns.

### Q3
Examine the shape of your data using the .shape command and the data types of your columns using .dtypes.

In [4]:
print(
    f'Shape: {df1.shape}\n'
    f'Data types: \n{df1.dtypes}'
)

Shape: (8719, 15)
Data types: 
Unnamed: 0      int64
origin         object
year            int64
month         float64
day           float64
hour          float64
temp          float64
dewp          float64
humid         float64
wind_dir      float64
wind_speed    float64
wind_gust     float64
precip        float64
pressure      float64
visib         float64
dtype: object


The `shape` attribute returns the number of rows (8719) and columns (15). The `dtypes` attribute returns the data type of each column. Most columns are floats; the index and year columns are integers; and the origin column is an object (which as I understand, is a more efficient way of storing strings in pandas dataframes.

### Q4
Use .describe() on your data. What do you notice about your data? What does this command return?

In [5]:
df1.describe()

Unnamed: 0    year        month          day         hour  \
count  8719.000000  8719.0  8718.000000  8718.000000  8718.000000   
mean   4360.000000  2013.0     6.506538    15.679628    11.515715   
std    2517.102832     0.0     3.439961     8.768250     6.916846   
min       1.000000  2013.0     1.000000     1.000000     0.000000   
25%    2180.500000  2013.0     4.000000     8.000000     6.000000   
50%    4360.000000  2013.0     7.000000    16.000000    12.000000   
75%    6539.500000  2013.0     9.000000    23.000000    18.000000   
max    8719.000000  2013.0    12.000000    31.000000    23.000000   

              temp         dewp        humid     wind_dir   wind_speed  \
count  8718.000000  8718.000000  8718.000000  8486.000000  8718.000000   
mean     55.484942    41.794184    62.863333   193.194674     9.335732   
std      18.345817    19.619776    19.572699   110.541333    12.328261   
min      10.940000    -9.040000    13.950000     0.000000     0.000000   
25%      39.920000    26.060000    47.602500   110.000000     5.753900   
50%      55.940000    42.800000    62.330000   220.000000     9.206240   
75%      71.060000    57.920000    78.790000   280.000000    12.658580   
max     100.040000    75.920000   100.000000   360.000000  1048.360580   

         wind_gust       precip     pressure        visib  
count  8718.000000  8719.000000  7780.000000  8719.000000  
mean     10.743374     0.003035  1017.826465     9.224522  
std      14.187117     0.022686     7.396722     2.094044  
min       0.000000     0.000000   983.900000     0.120000  
25%       6.621473     0.000000  1012.800000    10.000000  
50%      10.594357     0.000000  1017.500000    10.000000  
75%      14.567241     0.000000  1022.900000    10.000000  
max    1206.432388     1.180000  1041.900000    10.000000

The data is primarily numeric, as 14 of the 15 columns were returned by `.describe()`. There appear to be relatively few missing values, as most of the columns have the same count of 8718. It's also clear that all of the data is from year 2013.

### Q5
Use the .head() and .tail() command - what does this do?

In [6]:
df1.head()

Unnamed: 0 origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
0           1    EWR  2013    1.0  1.0   0.0  37.04  21.92  53.97     230.0   
1           2    EWR  2013    1.0  1.0   1.0  37.04  21.92  53.97     230.0   
2           3    EWR  2013    1.0  1.0   2.0  37.94  21.92  52.09     230.0   
3           4    EWR  2013    1.0  1.0   3.0  37.94  23.00  54.51     230.0   
4           5    EWR  2013    1.0  1.0   4.0  37.94  24.08  57.04     240.0   

   wind_speed  wind_gust  precip  pressure  visib  
0    10.35702  11.918651     0.0    1013.9   10.0  
1    13.80936  15.891535     0.0    1013.0   10.0  
2    12.65858  14.567241     0.0    1012.6   10.0  
3    13.80936  15.891535     0.0    1012.7   10.0  
4    14.96014  17.215830     0.0    1012.8   10.0

In [7]:
df1.tail()

Unnamed: 0 origin  year  month   day  hour   temp   dewp  humid  \
8714        8715    JFK  2013    9.0   2.0  20.0  75.20  73.40  94.14   
8715        8716    JFK  2013   10.0  23.0  10.0  48.92  39.02  68.51   
8716        8717    JFK  2013   10.0  23.0  11.0  48.92  39.02  68.51   
8717        8718    JFK  2013   12.0  17.0   5.0  26.96  10.94  50.34   
8718        8719    LGA  2013    8.0  22.0  22.0  75.92  66.92  73.68   

      wind_dir  wind_speed  wind_gust  precip  pressure  visib  
8714     200.0     4.60312   5.297178     0.0       NaN    4.0  
8715      60.0     4.60312   5.297178     0.0    1008.1   10.0  
8716      40.0     4.60312   5.297178     0.0    1008.5   10.0  
8717      40.0     4.60312   5.297178     0.0    1023.9   10.0  
8718     210.0     8.05546   9.270062     0.0    1011.9   10.0

The `.head(n)` command returns the first $n$ rows, with $n=5$ as the default. The `.tail(n)` command returns the the *last* $n$ rows, with $n=5$ as the default. 

# Dataset 2 - Salaries

### Q1
Load your dataset into python using the pandas library and save data into a dataframe named dfi (where i is one of your datasets, for a total of 4).

In [8]:
df2 = pd.read_csv('https://github.com/data602sps/datasetspractice/raw/main/salarydata.csv')

### Q2
Preview your data by calling your dataframe’s name. How many columns and rows do you see?

In [9]:
df2

Sate  2018 Mean Annual Wage  Unnamed: 2
0               Alabama                 90,620         NaN
1                 Alaska                99,180         NaN
2                Arizona                88,800         NaN
3               Arkansas                81,430         NaN
4             California                97,110         NaN
5               Colorado                82,560         NaN
6            Connecticut                96,010         NaN
7               Delaware                93,880         NaN
8   District of Columbia                89,880         NaN
9                Florida                87,410         NaN
10               Georgia                86,320         NaN
11                Hawaii                90,540         NaN
12                 Idaho                77,700         NaN
13              IIIinois                90,690         NaN
14               lndiana                83,680         NaN
15                  lowa                82,960         NaN
16                Kansas                85,250         NaN
17              Kentucky                84,630         NaN
18             Louisiana                89,860         NaN
19                 Maine                79,910         NaN
20              Maryland                85,170         NaN
21         Massachusetts                91,750         NaN
22              Michigan                91,160         NaN
23             Minnesota                83,750         NaN
24           Mississippi                89,720         NaN
25              Missouri                81,330         NaN
26               Montana                79,050         NaN
27              Nedraska                80,130         NaN
28                Nevada               107,920         NaN
29         New Hampshire                82,880         NaN
30            New Jersey                97,770         NaN
31           New  Mexico                97,210         NaN
32              New York                87,470         NaN
33        North Carolina                87,560         NaN
34          North Dakota                78,120         NaN
35                  Ohio                86,690         NaN
36              Okahoma                 84,860         NaN
37                Oregon                85,890         NaN
38          Pennsylvania                87,050         NaN
39          Rhode Island                83,850         NaN
40        South Carolina                85,450         NaN
41          South Dakota                92,940         NaN
42             Tennessee                85,940         NaN
43                 Taxas                92,940         NaN
44                  Utah                85,940         NaN
45               Vermont                75,010         NaN
46              Virginia                91,700         NaN
47            Washington                85,930         NaN
48         West Virginia                89,420         NaN
49             Wisconsin                85,200         NaN
50               Wyomimg                87,510         NaN

Because this dataframe has fewer rows, so Jupyter doesn't truncate the display. All 50 rows are shown, along with 3 columns. The third column appears to be empty, likely the result of erroneous commas in the .csv file.

### Q3
Examine the shape of your data using the .shape command and the data types of your columns using .dtypes.

In [10]:
print(
    f'Shape: {df2.shape}\n'
    f'Data types: \n{df2.dtypes}'
)

Shape: (51, 3)
Data types: 
Sate                      object
2018 Mean Annual Wage     object
Unnamed: 2               float64
dtype: object


The `shape` attribute returns the number of rows (51) and columns (3). The `dtypes` attribute returns the data type of each column. The NaN column appears as a float, but the two "real" columns are dtype = object. While the State column appears to have been interpreted correctly, the Mean Wage column seems to have been incorrectly interpreted because of the thousands comma formatting.

### Q4
Use .describe() on your data. What do you notice about your data? What does this command return?

In [11]:
df2.describe()

Unnamed: 2
count         0.0
mean          NaN
std           NaN
min           NaN
25%           NaN
50%           NaN
75%           NaN
max           NaN

The `.describe()` function returns no valuable information, because of the aforementioned misinterpretation of the only numeric column (Mean Wage).

### Q5
Use the .head() and .tail() command - what does this do?

In [12]:
df2.head()

Sate  2018 Mean Annual Wage  Unnamed: 2
0    Alabama                 90,620         NaN
1      Alaska                99,180         NaN
2     Arizona                88,800         NaN
3    Arkansas                81,430         NaN
4  California                97,110         NaN

In [13]:
df2.tail()

Sate  2018 Mean Annual Wage  Unnamed: 2
46       Virginia                91,700         NaN
47     Washington                85,930         NaN
48  West Virginia                89,420         NaN
49      Wisconsin                85,200         NaN
50        Wyomimg                87,510         NaN

The `.head(n)` command returns the first $n$ rows, with $n=5$ as the default. The `.tail(n)` command returns the the *last* $n$ rows, with $n=5$ as the default. 

# Dataset 3 - TBD

### Q1
Load your dataset into python using the pandas library and save data into a dataframe named dfi (where i is one of your datasets, for a total of 4).

In [8]:
df2 = pd.read_csv('https://github.com/data602sps/datasetspractice/raw/main/salarydata.csv')

### Q2
Preview your data by calling your dataframe’s name. How many columns and rows do you see?

In [9]:
df2

Sate  2018 Mean Annual Wage  Unnamed: 2
0               Alabama                 90,620         NaN
1                 Alaska                99,180         NaN
2                Arizona                88,800         NaN
3               Arkansas                81,430         NaN
4             California                97,110         NaN
5               Colorado                82,560         NaN
6            Connecticut                96,010         NaN
7               Delaware                93,880         NaN
8   District of Columbia                89,880         NaN
9                Florida                87,410         NaN
10               Georgia                86,320         NaN
11                Hawaii                90,540         NaN
12                 Idaho                77,700         NaN
13              IIIinois                90,690         NaN
14               lndiana                83,680         NaN
15                  lowa                82,960         NaN
16                Kansas                85,250         NaN
17              Kentucky                84,630         NaN
18             Louisiana                89,860         NaN
19                 Maine                79,910         NaN
20              Maryland                85,170         NaN
21         Massachusetts                91,750         NaN
22              Michigan                91,160         NaN
23             Minnesota                83,750         NaN
24           Mississippi                89,720         NaN
25              Missouri                81,330         NaN
26               Montana                79,050         NaN
27              Nedraska                80,130         NaN
28                Nevada               107,920         NaN
29         New Hampshire                82,880         NaN
30            New Jersey                97,770         NaN
31           New  Mexico                97,210         NaN
32              New York                87,470         NaN
33        North Carolina                87,560         NaN
34          North Dakota                78,120         NaN
35                  Ohio                86,690         NaN
36              Okahoma                 84,860         NaN
37                Oregon                85,890         NaN
38          Pennsylvania                87,050         NaN
39          Rhode Island                83,850         NaN
40        South Carolina                85,450         NaN
41          South Dakota                92,940         NaN
42             Tennessee                85,940         NaN
43                 Taxas                92,940         NaN
44                  Utah                85,940         NaN
45               Vermont                75,010         NaN
46              Virginia                91,700         NaN
47            Washington                85,930         NaN
48         West Virginia                89,420         NaN
49             Wisconsin                85,200         NaN
50               Wyomimg                87,510         NaN

In PyCharm, 10 rows and 15 columns are displayed.

### Q3
Examine the shape of your data using the .shape command and the data types of your columns using .dtypes.

In [10]:
print(
    f'Shape: {df2.shape}\n'
    f'Data types: \n{df2.dtypes}'
)

Shape: (51, 3)
Data types: 
Sate                      object
2018 Mean Annual Wage     object
Unnamed: 2               float64
dtype: object


The `shape` attribute returns the number of rows (8719) and columns (15). The `dtypes` attribute returns the data type of each column. Most columns are floats; the index and year columns are integers; and the origin column is an object (which as I understand, is a more efficient way of storing strings in pandas dataframes.

### Q4
Use .describe() on your data. What do you notice about your data? What does this command return?

In [11]:
df2.describe()

Unnamed: 2
count         0.0
mean          NaN
std           NaN
min           NaN
25%           NaN
50%           NaN
75%           NaN
max           NaN

The data is primarily numeric, as 14 of the 15 columns were returned by `.describe()`. There appear to be relatively few missing values, as most of the columns have the same count of 8718. It's also clear that all of the data is from year 2013.

### Q5
Use the .head() and .tail() command - what does this do?

In [12]:
df2.head()

Sate  2018 Mean Annual Wage  Unnamed: 2
0    Alabama                 90,620         NaN
1      Alaska                99,180         NaN
2     Arizona                88,800         NaN
3    Arkansas                81,430         NaN
4  California                97,110         NaN

In [13]:
df2.tail()

Sate  2018 Mean Annual Wage  Unnamed: 2
46       Virginia                91,700         NaN
47     Washington                85,930         NaN
48  West Virginia                89,420         NaN
49      Wisconsin                85,200         NaN
50        Wyomimg                87,510         NaN

The `.head(n)` command returns the first $n$ rows, with $n=5$ as the default. The `.tail(n)` command returns the the *last* $n$ rows, with $n=5$ as the default. 

# Dataset 4 - TBD

### Q1
Load your dataset into python using the pandas library and save data into a dataframe named dfi (where i is one of your datasets, for a total of 4).

In [8]:
df2 = pd.read_csv('https://github.com/data602sps/datasetspractice/raw/main/salarydata.csv')

### Q2
Preview your data by calling your dataframe’s name. How many columns and rows do you see?

In [9]:
df2

Sate  2018 Mean Annual Wage  Unnamed: 2
0               Alabama                 90,620         NaN
1                 Alaska                99,180         NaN
2                Arizona                88,800         NaN
3               Arkansas                81,430         NaN
4             California                97,110         NaN
5               Colorado                82,560         NaN
6            Connecticut                96,010         NaN
7               Delaware                93,880         NaN
8   District of Columbia                89,880         NaN
9                Florida                87,410         NaN
10               Georgia                86,320         NaN
11                Hawaii                90,540         NaN
12                 Idaho                77,700         NaN
13              IIIinois                90,690         NaN
14               lndiana                83,680         NaN
15                  lowa                82,960         NaN
16                Kansas                85,250         NaN
17              Kentucky                84,630         NaN
18             Louisiana                89,860         NaN
19                 Maine                79,910         NaN
20              Maryland                85,170         NaN
21         Massachusetts                91,750         NaN
22              Michigan                91,160         NaN
23             Minnesota                83,750         NaN
24           Mississippi                89,720         NaN
25              Missouri                81,330         NaN
26               Montana                79,050         NaN
27              Nedraska                80,130         NaN
28                Nevada               107,920         NaN
29         New Hampshire                82,880         NaN
30            New Jersey                97,770         NaN
31           New  Mexico                97,210         NaN
32              New York                87,470         NaN
33        North Carolina                87,560         NaN
34          North Dakota                78,120         NaN
35                  Ohio                86,690         NaN
36              Okahoma                 84,860         NaN
37                Oregon                85,890         NaN
38          Pennsylvania                87,050         NaN
39          Rhode Island                83,850         NaN
40        South Carolina                85,450         NaN
41          South Dakota                92,940         NaN
42             Tennessee                85,940         NaN
43                 Taxas                92,940         NaN
44                  Utah                85,940         NaN
45               Vermont                75,010         NaN
46              Virginia                91,700         NaN
47            Washington                85,930         NaN
48         West Virginia                89,420         NaN
49             Wisconsin                85,200         NaN
50               Wyomimg                87,510         NaN

In PyCharm, 10 rows and 15 columns are displayed.

### Q3
Examine the shape of your data using the .shape command and the data types of your columns using .dtypes.

In [10]:
print(
    f'Shape: {df2.shape}\n'
    f'Data types: \n{df2.dtypes}'
)

Shape: (51, 3)
Data types: 
Sate                      object
2018 Mean Annual Wage     object
Unnamed: 2               float64
dtype: object


The `shape` attribute returns the number of rows (8719) and columns (15). The `dtypes` attribute returns the data type of each column. Most columns are floats; the index and year columns are integers; and the origin column is an object (which as I understand, is a more efficient way of storing strings in pandas dataframes.

### Q4
Use .describe() on your data. What do you notice about your data? What does this command return?

In [11]:
df2.describe()

Unnamed: 2
count         0.0
mean          NaN
std           NaN
min           NaN
25%           NaN
50%           NaN
75%           NaN
max           NaN

The data is primarily numeric, as 14 of the 15 columns were returned by `.describe()`. There appear to be relatively few missing values, as most of the columns have the same count of 8718. It's also clear that all of the data is from year 2013.

### Q5
Use the .head() and .tail() command - what does this do?

In [12]:
df2.head()

Sate  2018 Mean Annual Wage  Unnamed: 2
0    Alabama                 90,620         NaN
1      Alaska                99,180         NaN
2     Arizona                88,800         NaN
3    Arkansas                81,430         NaN
4  California                97,110         NaN

In [13]:
df2.tail()

Sate  2018 Mean Annual Wage  Unnamed: 2
46       Virginia                91,700         NaN
47     Washington                85,930         NaN
48  West Virginia                89,420         NaN
49      Wisconsin                85,200         NaN
50        Wyomimg                87,510         NaN

The `.head(n)` command returns the first $n$ rows, with $n=5$ as the default. The `.tail(n)` command returns the the *last* $n$ rows, with $n=5$ as the default. 